src/express-http-server.js

```diff
@@ -13,18 +15,23 @@ class ExpressHTTPServer {
     this.password = options.password;
     this.cache = options.cache;
     this.gzip = options.gzip || false;
+    this.preFastbootMiddlewares = options.preFastbootMiddlewares || noop;
+    this.postFastbootMiddlewares = options.postFastbootMiddlewares || noop;
 
     this.app = express();
-    if (options.gzip) {
-      this.app.use(require('compression')());
-    }
   }
 
-  serve(middleware) {
+  serve(fastbootMiddleware) {
     let app = this.app;
     let username = this.username;
     let password = this.password;
 
+    this.preFastbootMiddlewares(app);
```

Comment:
@tomdale Since [`app.use`](http://expressjs.com/en/api.html#app.use) is synchronous way of binding callbacks all we need to do is pass the `app` instance into these hooks.